core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1123,8 +1133,15 @@ class DAGScheduler(
             // TODO: Cancel running tasks in the stage
             logInfo(s"Resubmitting $mapStage (${mapStage.name}) and " +
               s"$failedStage (${failedStage.name}) due to fetch failure")
+            // We might get lots of fetch failed for this stage, from lots of executors.
+            // Its better if we can resubmit for all the failed executors at one time, so lets
+            // just wait a *bit* before we resubmit.
```

Comment:
see https://github.com/apache/spark/pull/8560 to show that you can remove the messageScheduler completely (the tests pass, anyway ...)